### PR TITLE
Update libraries to newer versions.

### PR DESCRIPTION
**Pull request by akatsoulas**
* Fixes warnings about deprecated libs.
* Fixes an error with MySQL-Python
* Satisfies requirement dependencies.

### CODE REVIEW

**Comment by pmac** [comment]
Lots of test failures. Looks like something in the newer Jinja isn't happy with how we use it :/

**Comment by akatsoulas** [comment]
This change was introduced in version 2.8
http://jinja.pocoo.org/docs/2.10/changelog/#version-2-8

**Comment by akatsoulas** [comment]
@pmac this PR looks a bit better now. Quick r? when you have some time? Interested in your thoughts about the changes.

File: kitsune/notifications/api.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ def has_object_permission(self, request, view, obj):
 
 class NotificationSerializer(serializers.ModelSerializer):
     is_read = serializers.ReadOnlyField()
-    timestamp = DateTimeUTCField(source='action.timestamp')
+    timestamp = DateTimeUTCField('%Y-%m-%dT%H:%M:%SZ', source='action.timestamp')
```
Comment (pmac) [inline kitsune/notifications/api.py]: Why was this change necessary? Is this a fix from the previous upgrade of DRF?

Comment (akatsoulas) [reply kitsune/notifications/api.py]: I believe this is coming from the replacement of `MySQL-Python` with `mysqlclient`. Probably the former did not return any microseconds.

Comment (pmac) [reply kitsune/notifications/api.py]: Oh right. Interesting. Thanks.
